semconv/internal/http.go

```diff
@@ -210,7 +210,9 @@ func (sc *SemanticConventions) httpBasicAttributesFromHTTPRequest(request *http.
 // HTTPServerMetricAttributesFromHTTPRequest generates low-cardinality attributes
 // to be used with server-side HTTP metrics.
 func (sc *SemanticConventions) HTTPServerMetricAttributesFromHTTPRequest(serverName string, request *http.Request) []attribute.KeyValue {
-	attrs := []attribute.KeyValue{}
+	attrs := []attribute.KeyValue{
+		sc.HTTPMethodKey.String(request.Method),
```

Comment:
This should probably go in `httpBasicAttributesFromHTTPRequest()` instead, as it is an attribute that will be available and applicable regardless whether it is a client or server request.

Reply:
- httpBasicAttributesFromHTTPRequest is called by httpCommonAttributesFromHTTPRequest
- httpCommonAttributesFromHTTPRequest is called by HTTPClientAttributesFromHTTPRequest

In HTTPClientAttributesFromHTTPRequest, the method is already appended to the attributes slice, 

https://github.com/open-telemetry/opentelemetry-go/blob/5568a3072367bb8ac4d2d9e759e7deb5a975b9f5/semconv/internal/http.go#L150-L154

do you think I should remove the check of request.Method in HTTPClientAttributesFromHTTPRequest too?

Reply:
@Aneurysm9 I change as you said, now parse the http.method in `httpBasicAttributesFromHTTPRequest`.

I summarize the current implement to following function chain

for trace data, two function exist

- HTTPClientAttributesFromHTTPRequest: parse http.method
  - httpCommonAttributesFromHTTPRequest
    - httpBasicAttributesFromHTTPRequest : parse http.scheme/flavor/host

- HTTPServerAttributesFromHTTPRequest: parse http.method
   - httpCommonAttributesFromHTTPRequest
      - httpBasicAttributesFromHTTPRequest:  parse http.scheme/flavor/host

for metric data, one function exists

- HTTPServerMetricAttributesFromHTTPRequest: parse http.method
   - httpBasicAttributesFromHTTPRequest: parse http.scheme/flavor/host

In above three functions, we found `HTTPClientAttributesFromHTTPRequest`, `HTTPServerAttributesFromHTTPRequest` and `HTTPServerMetricAttributesFromHTTPRequest` all parse http.method, so we can move this logic into `httpBasicAttributesFromHTTPRequest`. Since in the [semantic_conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions) of span and metrics, the `http.method` are all required field.

I also add a test for `httpBasicAttributesFromHTTPRequest`.

Please help review this, thank you!